base/commands/migration/status_stages.go

```diff
@@ -0,0 +1,152 @@
+package migration
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/hazelcast/hazelcast-commandline-client/clc/ux/stage"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/output"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+	serialization2 "github.com/hazelcast/hazelcast-commandline-client/internal/serialization"
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/types"
+)
+
+type StatusStages struct {
+	migrationID              string
+	ci                       *hazelcast.ClientInternal
+	migrationsInProgressList *hazelcast.List
+	statusMap                *hazelcast.Map
+	updateTopic              *hazelcast.Topic
+	topicListenerID          types.UUID
+	updateMsgChan            chan UpdateMessage
+}
+
+func NewStatusStages() *StatusStages {
+	return &StatusStages{}
+}
+
+func (st *StatusStages) Build(ctx context.Context, ec plug.ExecContext) []stage.Stage {
+	return []stage.Stage{
+		{
+			ProgressMsg: "Connecting to the migration cluster",
+			SuccessMsg:  "Connected to the migration cluster",
+			FailureMsg:  "Could not connect to the migration cluster",
+			Func:        st.connectStage(ctx, ec),
+		},
+		{
+			ProgressMsg: "Fetching migration status",
+			SuccessMsg:  "Fetched migration status",
+			FailureMsg:  "Could not fetch migration status",
+			Func:        st.fetchStage(ctx, ec),
+		},
+	}
+}
+
+type MigrationInProgress struct {
+	MigrationID string `json:"migrationId"`
+}
+
+func (st *StatusStages) connectStage(ctx context.Context, ec plug.ExecContext) func(statuser stage.Statuser) error {
+	return func(status stage.Statuser) error {
+		var err error
+		st.ci, err = ec.ClientInternal(ctx)
+		if err != nil {
+			return err
+		}
+		st.migrationsInProgressList, err = st.ci.Client().GetList(ctx, MigrationsInProgressList)
+		if err != nil {
+			return err
+		}
+		all, err := st.migrationsInProgressList.GetAll(ctx)
+		if err != nil {
+			return err
+		}
+		m := all[0].(MigrationInProgress)
+		st.migrationID = m.MigrationID
+		st.statusMap, err = st.ci.Client().GetMap(ctx, MakeStatusMapName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateTopic, err = st.ci.Client().GetTopic(ctx, MakeUpdateTopicName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateMsgChan = make(chan UpdateMessage)
+		_, err = st.updateTopic.AddMessageListener(ctx, st.topicListener)
```

Comment:
Note: If the status is a terminal status, we should not add this message listener. Only if the migration is not completed yet.

Reply:
But we need to read at least one message to be able to know if the status is terminal or not.

Reply:
That should be done by reading the status map once. Not via update topic

Reply:
`__datamigrations_in_progress` list should return just the migrations in progress. If its status is terminal, `__datamigrations_in_progress` list should not contain it, right?

Reply:
Why is that? What is the difference between reading from status map and update topic?

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/367/commits/48a42ce85eb11a666198b6b903ac772cb1609562